src/main/java/org/gridsuite/modification/server/repositories/GeneratorModificationRepository.java

```diff
@@ -26,4 +28,16 @@ public interface GeneratorModificationRepository extends JpaRepository<Generator
 
     @EntityGraph(attributePaths = {"properties"}, type = EntityGraph.EntityGraphType.LOAD)
     List<GeneratorModificationEntity> findAllPropertiesByIdIn(List<UUID> ids);
+
+    @Modifying
+    @Query(value = "BEGIN;" +
+        "DELETE FROM generator_modification_entity_reactive_capability_curve_points cp WHERE cp.generator_modification_entity_id IN ?1 ;" +
+        "DELETE FROM free_property fp WHERE fp.equipment_modification_id IN ?1 ;" +
+        "DELETE FROM generator_modification WHERE id IN ?1 ;" +
+        "DELETE FROM tabular_modification_modifications WHERE tabular_modification_entity_id = ?2 ;" +
+        "DELETE FROM modification WHERE id IN ?1 ;" +
+        "DELETE FROM tabular_modification WHERE id = ?2 ;" +
+        "DELETE FROM modification WHERE id = ?2 ;" +
+        "COMMIT;", nativeQuery = true)
```

Comment:
`"DELETE FROM free_property fp WHERE fp.equipment_modification_id IN ?1 ;" +`
`"DELETE FROM tabular_modification_modifications WHERE tabular_modification_entity_id = ?2 ;" +`
`"DELETE FROM modification WHERE id IN ?1 ;" +`
`"DELETE FROM tabular_modification WHERE id = ?2 ;" +`
`"DELETE FROM modification WHERE id = ?2 ;"`

will be duplicated for each equipment modification. Could we put it common but keep the transaction ?

Reply:
We'll see that when we do the generalization ticket, no ?